src/metaswitch/clearwater/config_manager/config_access.py

```diff
@@ -128,6 +144,19 @@ def _check_connection(self):
             raise EtcdConnectionFailed(
                 "etcd process not running at {}".format(location))
 
+    def _check_etcd_cluster_health(self):
+        """Raises an EtcdNoQuorum exception if the etcd cluster does not have
+        quorum."""
+        try:
+            health_output = subprocess.check_output(self.ETCD_HEALTH_CHECK)
+        except subprocess.CalledProcessError:
+            # We couldn't even run the health check! something must be wrong.
+            raise EtcdNoQuorum()
+
+        if any(ind in health_output for ind in self.ETCD_UNHEALTHY_INDICATORS):
```

Comment:
I assume you've tried already, but just double checking in case you haven't :-) Can't we use any return codes from the script rather than reading out the output?

Reply:
This is the canonical way of doing this, so unfortunately we can't just use the return code output :(.